frontend/src/components/Scratch/ScratchPreferencesModal.tsx

```diff
@@ -22,6 +23,19 @@ function DiffPrefs() {
                 />
                 Automatically compile on change
             </label>
+            <label className={styles.intPreference}>
+                <input
+                    type="range"
+                    min="50" max="2000" step="50" value={autoRecompileDelay}
+                    onChange={evt => setAutoRecompileDelay(evt.target.value)}
+                />
+                <input className={styles.numberInput}
+                    type="number"
+                    min="50" max="2000" step="50" value={autoRecompileDelay}
+                    onChange={evt => setAutoRecompileDelay(evt.target.value)}
+                />
```

Comment:
I think it would be better to implement this as a ContentEditable so there isn't whitespace between the number and "ms". See the scratch name component for an example of this

Reply:
That way, there are no arrows for increasing/decreasing the value by clicking. Which one is preferred?

Reply:
I don't think the arrows are particularly needed since we have that range input next to it anyway

Reply:
We might be able to make our own fancy `TimePeriodInput` component that includes the "ms" prefix such that you can type "5s" and have it be understood to mean 5000 ms

Reply:
I'm not sure how quick or polished you want this to be merged, but I pushed the changes introducing a `TimePeriodInput`-component. It does not support different time units yet, but it's a `ContentEditable`.

Reply:
Awesome, looks good. In retrospect I don't think we need to include the suffix parsing thing, it's a bit overkill unless we decide we want longer time period inputs elsewhere in decomp.me